spyderlib/plugins/variableexplorer.py

```diff
@@ -126,20 +137,20 @@ def remove_shellwidget(self, shellwidget_id):
         # that shell was not a Python-based console (it was a terminal)
         if shellwidget_id in self.shellwidgets:
             nsb = self.shellwidgets.pop(shellwidget_id)
-            self.removeWidget(nsb)
+            self.stack.removeWidget(nsb)
             nsb.close()
     
     def set_shellwidget_from_id(self, shellwidget_id):
         if shellwidget_id in self.shellwidgets:
             nsb = self.shellwidgets[shellwidget_id]
-            self.setCurrentWidget(nsb)
+            self.stack.setCurrentWidget(nsb)

```

Comment:
Should I make accesors for the places where `stack` is called... so that all the calls are made to the variable explorer object and inside I make the correct call?

Like I did with `self.current_widget`?